### PR TITLE
Fix errors that occur crossing the queue boundary that cause uncaught exceptions

### PR DESCRIPTION
**Pull request by robskillington**
cc: @Raynos @jwolski @sh1mmer @kriskowal

### CODE REVIEW

**Comment by kriskowal** [comment]
This seems consistent with the intended semantics of an ephemeral socket. :+1:

File: lib/EphemeralSocket.js [hunk 1]
```diff
@@ -179,6 +179,11 @@ function allocSocket(self) {
     // When we error we should just close and de-allocate.
     function onError(err) {
         //TODO logger.error() about socket error
+
+        // First attach an error listener so consequent error
+        // events do not cause uncaught exceptions on this socket
+        self._socket.on('error', function() {});
```
Comment (Raynos) [inline lib/EphemeralSocket.js]: can we name this function and add a //TODO comment about logging inside the error event listeners

**Comment by Raynos** [comment]
lgtm.

File: lib/EphemeralSocket.js (hunk 1 above)
Comment (robskillington) [inline lib/EphemeralSocket.js]: For sure, sounds good

**Pull request merged**